apps/webapp/src/state/staking.ts

```diff
@@ -219,6 +220,43 @@ export const createStakingSlice = (): SliceCreator<StakingSlice> => (set, get) =
       }
     }
   },
+  undelegate: async () => {
+    const toast = new TransactionToast('undelegate');
+    toast.onStart();
+
+    try {
+      const transactionPlan = await plan(assembleUndelegateRequest(get().staking));
+
+      // Reset form _after_ building the transaction form, since it depends on
+      // the state.
+      set(state => {
+        state.staking.amount = '';
+        state.staking.action = undefined;
+        state.staking.validatorInfo = undefined;
+      });
+
+      const transaction = await authWitnessBuild({ transactionPlan }, status =>
+        toast.onBuildStatus(status),
+      );
+      const txHash = await getTxHash(transaction);
+      toast.txHash(txHash);
+      const { detectionHeight } = await broadcast({ transaction, awaitDetection: true }, status =>
+        toast.onBroadcastStatus(status),
+      );
+      toast.onSuccess(detectionHeight);
+
+      // Reload delegation tokens and unstaked tokens to reflect their updated
+      // balances.
+      void get().staking.loadDelegationsForCurrentAccount();
+      void get().staking.loadUnstakedTokensByAccount();
```

Comment:
Will the `void` cause it to exit this try/catch and not be handled if errored? 🤔

Reply:
Nope — [void simply has the effect of making the statement evaluate to `undefined`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void), which our linter requires when we're not doing anything with a promised return value.